etna/transforms/decomposition/dft_based.py

```diff
@@ -192,7 +192,9 @@ def transform(self, ts: TSDataset) -> TSDataset:
 
         segment_components = pd.concat(segment_components, axis=1)
 
-        ts.add_columns_from_pandas(segment_components)
```

Comment:
Let's add some tests to catch such behavior in the future. It can be inspired by the example in the issue.